bigframes/session/executor.py

```diff
@@ -0,0 +1,329 @@
+    ):
+        if bigframes.options.compute.enable_multi_query_execution:
+            self._simplify_with_caching(array_value)
+
+        sql = self.to_sql(
+            array_value, ordered=ordered, col_id_overrides=col_id_overrides
+        )
+        job_config = bigquery.QueryJobConfig()
+        # TODO(swast): plumb through the api_name of the user-facing api that
+        # caused this query.
+        return self._run_execute_query(
+            sql=sql,
+            job_config=job_config,
+        )
+
+    def export(
+        self,
+        array_value,
+        col_id_overrides: Mapping[str, str],
+        destination: bigquery.TableReference,
+        if_exists: Literal["fail", "replace", "append"] = "fail",
+        cluster_cols: Sequence[str] = [],
+    ):
+        dispositions = {
+            "fail": bigquery.WriteDisposition.WRITE_EMPTY,
+            "replace": bigquery.WriteDisposition.WRITE_TRUNCATE,
+            "append": bigquery.WriteDisposition.WRITE_APPEND,
+        }
+        sql = self.to_sql(array_value, ordered=False, col_id_overrides=col_id_overrides)
+        job_config = bigquery.QueryJobConfig(
+            write_disposition=dispositions[if_exists],
+            destination=destination,
+            clustering_fields=cluster_cols if cluster_cols else None,
+        )
+        # TODO(swast): plumb through the api_name of the user-facing api that
+        # caused this query.
+        return self._run_execute_query(
+            sql=sql,
+            job_config=job_config,
+        )
+
+    def dry_run(self, array_value: bigframes.core.ArrayValue, ordered: bool = True):
+        sql = self.to_sql(array_value, ordered=ordered)
+        job_config = bigquery.QueryJobConfig(dry_run=True)
+        query_job = self.bqclient.query(sql, job_config=job_config)
+        results_iterator = query_job.result()
+        return results_iterator, query_job
+
+    def peek(
+        self, array_value: bigframes.core.ArrayValue, n_rows: int
+    ) -> tuple[bigquery.table.RowIterator, bigquery.QueryJob]:
+        """A 'peek' efficiently accesses a small number of rows in the dataframe."""
+        if not tree_properties.peekable(self._with_cached_executions(array_value.node)):
+            warnings.warn("Peeking this value cannot be done efficiently.")
+        sql = self.compiler.compile_peek(
+            self._with_cached_executions(array_value.node), n_rows
+        )
+
+        # TODO(swast): plumb through the api_name of the user-facing api that
+        # caused this query.
+        return self._run_execute_query(
+            sql=sql,
+        )
+
+    # Helpers
+    def _run_execute_query(
+        self,
+        sql: str,
+        job_config: Optional[bq_job.QueryJobConfig] = None,
+        api_name: Optional[str] = None,
+    ) -> Tuple[bigquery.table.RowIterator, bigquery.QueryJob]:
+        """
+        Starts BigQuery query job and waits for results.
+        """
+        job_config = bigquery.QueryJobConfig()
```

Comment:
The `job_config` overwrites the one in the function parameter. Could you double check if it's in purpose.

Reply:
yeah, we don't want that. fixed in new revision